app/javascript/components/student/Editor.tsx

```diff
@@ -233,25 +243,32 @@ export function Editor({
           controllerRef.current = undefined
         })
     },
-    [abort, controllerRef]
+    [abort, controllerRef, isMountedRef]
   )
   const submit = useCallback(
     (e: React.MouseEvent<HTMLButtonElement>) => {
       if (!submission) {
         return
       }
 
+      controllerRef.current = new AbortController()
+
       dispatch({ type: ActionType.CREATING_ITERATION })
 
       fetchJSON(submission.links.submit, {
         method: 'POST',
         body: JSON.stringify({}),
+        signal: controllerRef.current.signal,
       }).then((json: any) => {
+        if (!isMountedRef.current) {
+          return
+        }
+
         const iteration = typecheck<Iteration>(json, 'iteration')
         location.assign(iteration.links.self)
       })
     },
-    [submission]
+    [controllerRef, isMountedRef, submission]
```

Comment:
Do we also need to add the following code here:

```
.finally(() => {
          controllerRef.current = undefined
        })
```

This is used in the above `fetchJSON` call.

Reply:
Yes. We need to :) Thanks.